hbase-client/src/main/java/org/apache/hadoop/hbase/exceptions/ClientExceptionsUtil.java

```diff
@@ -59,18 +58,22 @@ public static boolean isMetaClearingException(Throwable cur) {
     if (cur == null) {
       return true;
     }
-    return !isSpecialException(cur) || (cur instanceof RegionMovedException)
-        || cur instanceof NotServingRegionException;
+    return !regionDefinitelyOnTheRegionServerException(cur);
   }
 
-  public static boolean isSpecialException(Throwable cur) {
-    return (cur instanceof RegionMovedException || cur instanceof RegionOpeningException
-        || cur instanceof RegionTooBusyException || cur instanceof RpcThrottlingException
-        || cur instanceof MultiActionResultTooLarge || cur instanceof RetryImmediatelyException
-        || cur instanceof CallQueueTooBigException || cur instanceof CallDroppedException
-        || cur instanceof NotServingRegionException || cur instanceof RequestTooBigException);
+  private static boolean regionDefinitelyOnTheRegionServerException(Throwable t) {
+    return (t instanceof RegionTooBusyException || t instanceof RpcThrottlingException
+        || t instanceof RetryImmediatelyException || t instanceof CallQueueTooBigException
+        || t instanceof CallDroppedException || t instanceof RequestTooBigException);
```

Comment:
And for RequestTooBigException, it is a DoNotRetryIOException so it is fine to not clear meta, but for CallQueueTooBigException and CallDroppedException, what if the region is not on the region server, and the region server can not recover for a very long time? The client will stuck there?

Reply:
CallQueueTooBigException and CallDroppedException are exceptions which imply the server is overload, so if we clear the meta here, there will be many requests which cause the overload sending to the server where the meta is, which may cause other problems? And because those two exceptions are recoverable, the client may receive RegionMovedException etc if the region is moved after the load is back to normal level and then will clear the meta? Maybe both of them should inherit from DoNotRetryNowIOException or something like that?

Reply:
But at least they do not mean the region is on the region server right? So the method name here is confusing...
And I still have concerns about do not retry on these two exceptions. Maybe we should not clear meta everytime, but as I said above, what if the region server can not recover all the time but the region has been moved elsewhere? The client will have no chance to update the meta?

Reply:
Yes, the name is confusing and should be renamed.
Yes, it is really a problem that the client has no change to update the meta in that situation. The problem here is this function is just return a boolean which says Yes or No. It seems we should add more status to carry this information, like Integer or something?

Reply:
For now I think we can keep the method there, but move these two exceptions out to the isMetaClearingException method to keep the old behavior. Can open another issue to discuss how to better handle these two exceptions. What do you think?

Reply:
Agreed, let me open another issue.